### PR TITLE
fixed adding of empty new header when no more header fields available

### PR DESCRIPTION
**Pull request by lfryc**
this also removes some unused code

### CODE REVIEW

**Comment by lfryc** [comment]
@cianclarke could you please review?

File: Gruntfile.js [hunk 1]
```diff
@@ -44,7 +33,7 @@ module.exports = function(grunt) {
   grunt.loadNpmTasks('grunt-nodemon');
   grunt.loadNpmTasks('assemble-less');
   grunt.loadNpmTasks('grunt-bower-task');
-  grunt.registerTask('serve', ['bower', 'less', 'nodemon']);
+  grunt.registerTask('serve', ['bower', 'nodemon']);
   grunt.registerTask('test', ['jshint', 'fh:unit']);
   grunt.registerTask('default', ['bower', 'less', 'test']);
```
Comment (cianclarke) [inline Gruntfile.js:38]: Missed the less task here

File: Gruntfile.js [hunk 2]
```diff
@@ -44,7 +33,7 @@ module.exports = function(grunt) {
   grunt.loadNpmTasks('grunt-nodemon');
   grunt.loadNpmTasks('assemble-less');
```
Comment (cianclarke) [inline Gruntfile.js:34]: can probably loose this - thought it was in my commit actually but guess not!

File: public/js/request.model.js [hunk 3]
```diff
@@ -39,23 +34,5 @@ App.RequestModel = Backbone.Model.extend({
       return xhr.responseText;
     }
     return textStatus;
-  },
```
Comment (cianclarke) [inline public/js/request.model.js:42]: nice find

File: public/js/request.view.js [hunk 4]
```diff
@@ -212,7 +212,7 @@ App.RequestView = App.BaseMapperView.extend({
     var el = $(e.target),
     headerRow = el.parents('.headerRow');
     headerRow.remove();
-    if (!this.model.get('headers') || this.model.get('headers').length === 0){
+    if (this.$el.find('#editableHeaders ul li.headerRow').size() === 0){
```
Comment (cianclarke) [inline public/js/request.view.js:215]: Looks good but shouldn't be needed I thought, hm - odd.. Will ping you on iRC!

**Comment by cianclarke** [comment]
Looks great

**Pull request merged**